### ISSUE TITLE
[P2] Token EmbeddingとPositional Encodingを視覚化しアプリ内学習へ接続するデモを実装する

### ISSUE THREAD

**Issue by tomiya7688** [tomiya7688/comfyUI_support_tools#119]
## 目的
Transformer系デモの入口として、**tokenをベクトルへ変換するEmbedding**と、系列順序を与える**Positional Encoding**を、数値・図・学習挙動としてアプリ内で確認できるデモを実装する。

本Issueを含むTransformer系デモはすべて、**ビルド済みアプリ本体へ組み込み、アプリ内で学習・推論・可視化まで完結する**ことを前提とする。Python等を使う場合も補助バックエンドに限定し、アプリ本体の実装言語・既存UI/アーキテクチャは変更しない。

## ビルド成果物に関する制約
- Pythonは開発時の学習・変換・検証等に利用してよい
- **`.py` 等のPythonスクリプトを配布用ビルド成果物へ残さない**
- 配布アプリには、必要に応じて変換済みモデル・重み・設定・ネイティブ/既存runtimeから読める最小成果物だけを含める
- Python実行環境が配布アプリの必須条件にならない構成を優先する
- Python補助ツールを使う場合も、source tree / build pipeline側に留め、release packageから除外する
- ビルドテストで `.py`, `__pycache__`, `.pyc` 等が成果物へ混入していないことを検証する

## 実装内容
- [ ] token列を入力できる
- [ ] token idを表示
- [ ] embedding vectorを各tokenごとに表示
- [ ] embedding tableの該当rowをハイライト
- [ ] positional encodingを加算前/加算後で表示
- [ ] sinusoidal positional encodingを実装
- [ ] position indexを変更したときvectorが変わることを表示
- [ ] token順序を入れ替えたときposition情報が変わることを表示
- [ ] embedding dimensionを小さな値で変更可能にする
- [ ] 学習可能embeddingの場合、training stepでembedding値が更新されることを表示
- [ ] Resetで初期状態へ戻せる

## 視覚フロー
```text
Token
  ↓
Token ID
  ↓
Embedding Vector
  +
Positional Encoding
  ↓
Transformer Input Vector
```

## 教育用説明
- 「Embeddingはtokenを学習可能な数値vectorへ変換します」
- 「Self-Attention単体では順序を直接持たないため、位置情報を加えます」
- 「同じtokenでも位置が違えば入力vectorは変わります」

## 学習機構
- [ ] 小規模taskでembedding parameterを実際に更新できる
- [ ] update前後のembedding vectorを比較表示
- [ ] loss historyを表示
- [ ] optimizer / learning rateは共通学習基盤を再利用

## テスト
- [ ] 同じtoken idは同じembedding rowを参照する
- [ ] 異なるpositionでpositional encodingが変化する
- [ ] embedding + positional encodingの和が表示値と一致する
- [ ] 学習stepでembedding parameterが更新される
- [ ] アプリ本体がPython等の補助依存なしでも起動できる
- [ ] 配布用buildに `.py` / `.pyc` / `__pycache__` が含まれない

## 完了条件
- token→embedding→位置情報付与までをアプリ内で追える
- 学習によりembeddingが変化することを確認できる
- Self-Attentionへの入力がどのように作られるか理解できる
- 配布アプリにPythonスクリプトが残らない

## 依存
- #115 RNN言語モデル
- #117 Attention

## Out of Scope
- BPE/SentencePieceの本格実装
- 大規模語彙
- pretrained embedding配布
- Transformer本体
